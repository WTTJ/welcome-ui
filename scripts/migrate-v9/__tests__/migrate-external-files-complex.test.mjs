import { readFileSync } from 'fs'
import { resolve } from 'path'

import { beforeEach, describe, expect, it } from 'vitest'

import { copyDirSync, deleteDirRecursive } from '../helpers/file-utils.mjs'
import { setupExternalMigrationTest } from '../helpers/test-setup.mjs'
import { migrateAll } from '../index.mjs'

describe('External Migration - Complex Styled Components', () => {
  // Don't perform cleanup for now so that we can inspect the migrated files
  const tempDir = setupExternalMigrationTest(
    'ExternalComplexComponent-test',
    'ExternalComplexComponent'
  )

  const fixturesDir = resolve(
    __dirname,
    '..',
    '__tests__',
    '__fixtures__',
    'ExternalComplexComponent'
  )

  beforeEach(() => {
    // Clean up any existing temp directory and set up fresh copy
    deleteDirRecursive(tempDir)
    copyDirSync(fixturesDir, tempDir)
  })

  it('migrates complex styled components (styled(Component), template literals, interpolations)', async () => {
    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'index.tsx'), 'utf8')
    const generatedScss = readFileSync(resolve(tempDir, 'styles.module.scss'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('complex-component.tsx')
    expect(generatedScss).toMatchSnapshot('complex-styles.module.scss')
  })
})
