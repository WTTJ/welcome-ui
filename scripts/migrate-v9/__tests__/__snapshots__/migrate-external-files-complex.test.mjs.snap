// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`External Migration - Complex Styled Components > migrates complex styled components (styled(Component), template literals, interpolations) > complex-component.tsx 1`] = `
"import { FormattedMessage } from 'react-intl'
import { Text } from 'welcome-ui/Text'
import { classNames } from 'welcome-ui/utils'

import styles from './styles.module.scss'
const cx = classNames(styles)
type Variant = 'primary' | 'secondary'
export const Complex = ({ variant = 'primary' }: { variant?: Variant }) => {
  return (
    <div
      className={cx('wrapper')}
      style={
        {
          '--wrapper-background-color':
            variant === 'primary' ? 'var(--primary-500)' : 'var(--secondary-500)',
          '--wrapper-position': variant === 'primary' ? 'var(--absolute)' : 'var(--relative)',
        } as React.CSSProperties
      }
      variant={variant}
    >
      <Link className={cx('card')} elevated to="/fishcakes">
        <div className={cx('title-wrapper')}>
          <Text
            as="h3"
            className="text-neutral-10"
            className={cx('title')}
            displayDetail
            style={
              {
                '--title-margin-top': displayDetail ? 'var(--spacing-sm)' : '0',
              } as React.CSSProperties
            }
          >
            Complex component
          </Text>
        </div>
      </Link>
      <Text as="span" className="mt-auto text-center text-dark-50 text-sm" variant="h5">
        <FormattedMessage defaultMessage="Â© 2024 Acme Corp. All rights reserved." id="8XKXo+" />
      </Text>
      <button className={cx('trigger-button', 'is-active')}>Toggle</button>
    </div>
  )
}
"
`;

exports[`External Migration - Complex Styled Components > migrates complex styled components (styled(Component), template literals, interpolations) > complex-styles.module.scss 1`] = `
"/* Imported utilities */
@import 'welcome-ui/utils/theme/breakpoints';
@import 'welcome-ui/utils/theme/texts';

/* Generated SCSS mixins from CSS template literals */
@mixin trigger-active-styles {
    /* WUI V9 TO MIGRATE */
  /*
  \${Box} {
      background-color: var(--color-beige-10);
    }
  */

    color: var(--color-neutral-50);
    background-color: var(--color-beige-30);

    /* WUI V9 TO MIGRATE */
  /*
  \${OrganizationLogo} {
      outline-color: var(--color-beige-30) !important;
      border: 1px solid;
      border-color: var(--color-beige-50);
    }
  */

    /* WUI V9 TO MIGRATE */
  /*
  \${OrganizationName} {
      color: var(--color-neutral-90);
    }
  */

      }

@layer components {
.wrapper {
padding: var(--spacing-xxl) 0;
    background-color: var(--wrapper-background-color);
    isolation: isolate;
    position: var(--wrapper-position);
    min-height: calc(100vh - TOPNAV_HEIGHT /* WUI V9 TO MIGRATE: constant TOPNAV_HEIGHT */);

    @media screen and (min-width: breakpoints.$breakpoint-lg) {
      padding: var(--spacing-3xl) 0;
      position: var(--wrapper-position);
      margin: var(--spacing-sm);
    }
}

.card {
background: white;
  border-radius: var(--spacing-md);
  padding: var(--spacing-lg);
  
  &.elevated {
box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  }
}

.title {
/* WUI V9 TO MIGRATE */
@include texts.h4;
  color: var(--color-neutral-90);
  margin-top: var(--title-margin-top);
}

.title-wrapper {
display: flex;
  align-items: center;
  justify-content: center;
}

.organization-name {
color: var(--color-neutral-60);
  font-weight: var(--font-weight-medium);
  margin: 0;
  transition: color 300ms;
}

.trigger-button {
border: none;
  background-color: transparent;
  padding: var(--spacing-md) var(--spacing-lg);
  cursor: pointer;
  width: 100%;
  border-radius: var(--spacing-md);
  transition: background-color 300ms;
  bottom: var(--spacing-md);

  &.is-active {
display: flex;
      align-items: center;
      justify-content: center;
      padding: 0;
      height: 48px;
      width: 48px;
  }

  &:hover {
    @include trigger-active-styles;

    /* WUI V9 TO MIGRATE */
/*
\${OrganizationName} {
      color: var(--color-beige-80);
    }
*/
}

  &.is-active {
@include trigger-active-styles;
  };

  &:active {
    @include trigger-active-styles;
  }
}
}"
`;
